syntax = "proto3";

package chat;

import "chat.proto";
import "heartbeat.proto";

service ProxyService {
  // Client joins the chatroom
  rpc Join(JoinRequest) returns (JoinResponse);

  // Client sends a message to the server
  rpc SendMessage(Message) returns (Empty);

  // Client receives chat messages as a stream
  rpc ReceiveMessages(Empty) returns (stream Message);

  // Proxy receives switch server request from the heartbeat service
  rpc SwitchServer(SwitchRequest) returns (SwitchResponse);
}
